src/acvp_ecdsa.c

```diff
@@ -0,0 +1,324 @@
+    }
+    
+    return ACVP_SUCCESS;
+}
+
+
+/*
+ * This function simply releases the data associated with
+ * a test case.
+ */
+
+static ACVP_RESULT acvp_ecdsa_release_tc (ACVP_ECDSA_TC *stc) {
+    if (stc->curve) { free(stc->curve); }
+    if (stc->secret_gen_mode) { free(stc->secret_gen_mode); }
+    if (stc->hash_alg) { free(stc->hash_alg); }
+    if (stc->qy) { free(stc->qy); }
+    if (stc->qx) { free(stc->qx); }
+    if (stc->d) { free(stc->d); }
+    if (stc->r) { free(stc->r); }
+    if (stc->s) { free(stc->s); }
+    if (stc->message) { free(stc->message); }
+    
+    return ACVP_SUCCESS;
+}
+
+static ACVP_RESULT acvp_ecdsa_init_tc (ACVP_CTX *ctx,
+                                              ACVP_CIPHER cipher,
+                                              ACVP_ECDSA_TC *stc,
+                                              unsigned int tc_id,
+                                              char *curve,
+                                              char *secret_gen_mode,
+                                              char *qx,
+                                              char *qy,
+                                              char *message,
+                                              char *r,
+                                              char *s
+) {
+    memset(stc, 0x0, sizeof(ACVP_ECDSA_TC));
+    
+    stc->tc_id = tc_id;
+    stc->cipher = cipher;
+    
+    stc->curve = calloc(5, sizeof(char));
+    if (!stc->curve) { return ACVP_MALLOC_FAIL; }
+    strncpy(stc->curve, curve, strnlen(curve, 5));
+    
+    stc->secret_gen_mode = calloc(18, sizeof(char));
+    if (!stc->secret_gen_mode) { return ACVP_MALLOC_FAIL; }
+    strncpy(stc->secret_gen_mode, secret_gen_mode, strnlen(secret_gen_mode, 18));
+    
+    stc->qy = calloc(ACVP_RSA_EXP_LEN_MAX, sizeof(char));
+    if (!stc->qy) { return ACVP_MALLOC_FAIL; }
+    stc->qx = calloc(ACVP_RSA_EXP_LEN_MAX, sizeof(char));
+    if (!stc->qx) { return ACVP_MALLOC_FAIL; }
+    stc->r = calloc(ACVP_RSA_EXP_LEN_MAX, sizeof(char));
+    if (!stc->r) { return ACVP_MALLOC_FAIL; }
+    stc->s = calloc(ACVP_RSA_EXP_LEN_MAX, sizeof(char));
+    if (!stc->s) { return ACVP_MALLOC_FAIL; }
+    
+    if (cipher == ACVP_ECDSA_KEYVER || cipher == ACVP_ECDSA_SIGVER) {
+        strncpy((char *)stc->qx, qx, strnlen(qx, 128));
+        strncpy((char *)stc->qy, qy, strnlen(qy, 128));
+    }
+    if (cipher == ACVP_ECDSA_SIGVER) {
+        strncpy((char *)stc->s, s, strnlen(s, 128));
+        strncpy((char *)stc->r, r, strnlen(r, 128));
+    }
+    if (cipher == ACVP_ECDSA_SIGVER || cipher == ACVP_ECDSA_SIGGEN) {
+        stc->message = calloc(ACVP_RSA_EXP_LEN_MAX, sizeof(char));
+        if (!stc->message) { return ACVP_MALLOC_FAIL; }
+        strncpy((char *)stc->message, message, strnlen(message, ACVP_RSA_MSGLEN_MAX));
+    }
+    stc->d = calloc(ACVP_RSA_EXP_LEN_MAX, sizeof(char));
+    if (!stc->d) { return ACVP_MALLOC_FAIL; }
```

Comment:
Do we need to spit out an error message on the failure so we know what type it is or does the caller do that ?

Reply:
I will add more verbose logs here :)